src/OrbitGl/TrackManager.h

```diff
@@ -32,16 +32,18 @@
 #include "Viewport.h"
 
 class OrbitApp;
-class TimeGraph;
 
 namespace orbit_gl {
 
+class TrackContainer;
```

Comment:
Please not forward declarations.

Reply:
Is there a cyclic dependency? If yes then we should probably change the design.

Reply:
(I guess the usage of TrackContainer down below could be replaced by CaptureViewElement, couldn't it?)

Reply:
At least the implementation of `TrackManager` relies on `track_container_` being of type `TrackContainer*`, not `CaptureViewElement` (e.g. because it needs `VerticallyMoveIntoView`).

I'd argue it's not really introduced in this PR, but it actually replaces the previous forward declaration of `TimeGraph` with this one, which is at least of a much more narrow scope. As I see it it's mainly used due to `UpdateMovingTrackPositionInVisibleTracks`, and there is already a TODO to move this into `TrackContainer`. This makes sense to me, then the forward declaration should also be gone.

So IMO it's ok to keep it here, it's slightly better than before and will hopefully be tackled in the future.

Reply:
+1 with what Florian said.

I wanted to tackle it first, but this means moving the ordering logic to TrackContainer and IMO maintain the visibility in TrackManager, so we should alter the order of Ordering and filtering which shouldn't be a big issue but not a priority so I introduced a new bug last week: [_"// TODO(b/214280810): Move it to TrackContainer, as well as everything related with Ordered Tracks."_ ](https://github.com/google/orbit/blob/main/src/OrbitGl/TrackManager.cpp#L241).

Reply:
Understood. That makes sense to me!